src/workloads/execution/SSBColumnStoreIndex.yml

```diff
@@ -0,0 +1,640 @@
+        }
+      },
+      {
+        "$lookup": {
+          "from": "supplier",
+          "localField": "l_suppkey",
+          "foreignField": "s_suppkey",
+          "as": "supplier"
+        }
+      },
+      {"$addFields": {"supplier": {"$first": "$supplier"}}},
+      {"$match": {"supplier.nation.region.r_name": "AMERICA"}},
+      {
+        "$lookup": {
+          "from": "customer",
+          "localField": "order.o_custkey",
+          "foreignField": "c_custkey",
+          "as": "customer"
+        }
+      },
+      {"$addFields": {"customer": {"$first": "$customer"}}},
+      {"$match": {"customer.nation.region.r_name": "AMERICA"}},
+      {
+        "$group": {
+          "_id": {
+            "order_year": {"$year": "$order.o_orderdate"},
+            "c_nation": "$customer.nation.n_name"
+          },
+          "profit": {
+            "$sum": {
+              "$subtract": [
+                "$l_extendedprice",
+                {"$multiply": ["$l_extendedprice", "$l_discount"]}
+              ]
+            }
+          }
+        }
+      },
+      {"$addFields": {"order_year": "$_id.order_year", "customer_nation": "$_id.c_nation"} }
+    ]
+
+Q4_3Query: &Q4_3Query
+  Pipeline:
+    [
+      { $limit: 10000 },
+      {
+        "$match": {
+          "part.p_mfgr": "Manufacturer#1",
+          "order.o_orderdate": {
+            "$gte": ISODate("1997-01-01T00:00:00.000Z"),
+            "$lt": ISODate("1999-01-01T00:00:00.000Z")
+          }
+        }
+      },
+      {
+        "$lookup": {
+          "from": "supplier",
+          "localField": "l_suppkey",
+          "foreignField": "s_suppkey",
+          "as": "supplier"
+        }
+      },
+      {"$addFields": {"supplier": {"$first": "$supplier"}}},
+      {"$match": {"supplier.nation.n_name": "UNITED STATES"}},
+      {
+        "$lookup": {
+          "from": "customer",
+          "localField": "order.o_custkey",
+          "foreignField": "c_custkey",
+          "as": "customer"
+        }
+      },
+      {"$addFields": {"customer": {"$first": "$customer"}}},
+      {"$match": {"customer.nation.region.r_name": "AMERICA"}},
+      {
+        "$group": {
+          "_id": {
+            "order_year": {"$year": "$order.o_orderdate"},
+            "c_nation": "$customer.nation.n_name",
+            "p_brand": "$part.p_brand"
+          },
+          "profit": {
+            "$sum": {
+              "$subtract": [
+                "$l_extendedprice",
+                {"$multiply": ["$l_extendedprice", "$l_discount"]}
+              ]
+            }
+          }
+        }
+      },
+      {
+        "$addFields": {
+          "order_year": "$_id.order_year",
+          "customer_nation": "$_id.c_nation",
+          "part_brand": "$_id.p_brand"
+        }
+      },
+      {"$sort": {"order_year": 1, "customer_nation": 1, "part_brand": 1}}
+    ]
+
+# Defines how many times each test operation is executed.
+TestRepeatCount: &TestRepeatCount 1
+MaxPhases: &MaxPhases 42
```

Comment:
Is it possible to compute in yaml? I mean, have a variable for NumberOfTests, then set MaxPhases to `&NumberOfQueries * 2 +  2` (or however one would arrive to the number 42). A similar question for the elements of the `Active` arrays to make it easier in the future to add/drop a query and make it easier to understand how the phases are related to each other.

Reply:
This is a fair point. I'm going to have to reorganize the phases again because I noticed that now it is impossible to distinguish the CSI and non-CSI metrics (they use the same name--woops). I'll see if I can make the phase numbers more intuitive though.

Reply:
See [edca461](https://github.com/mongodb/genny/pull/760/commits/edca461bdcf65f9eb2bbcbe7fa1306fc459b7016). It's more flat-footed about the phase numbers, which hopefully makes it easier to add queries in the future. It's still not ideal though.

Reply:
I wonder, whether having so many phases with a single actor in each is an indicator that Genny isn't the best tool for the testing we are trying to do here (this is not a critic of the patch! just thinking aloud)